pkg/storj/encryption.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storj
+
+// EncryptionScheme is the scheme and parameters used for encryption
+type EncryptionScheme struct {
+	Algorithm      EncryptionAlgorithm
+	EncryptedKey   EncryptedPrivateKey
+	EncryptedNonce EncryptedNonce
+}
+
+// EncryptionAlgorithm specifies an encryption algorithm
+type EncryptionAlgorithm byte
+
+// List of supported encryption algorithms
+const (
+	InvalidEncryptionAlgorithm = EncryptionAlgorithm(iota)
```

Comment:
This is the zero, right? I would expect to have `Unencrypted` as the zero.

Reply:
For the encryption and redundancy I want the caller to always explicitly specify. It's common to forget to include or specify such things, in this case the selection is important. We could later default it to some value, but I would rather fail in such cases.

Reply:
Who is going to set this? For the moment we set it internally in the client based on the configuration in `config.yaml`. If it is not configured then we default to AESGCM (which is `1`).

Are we going to expose in the API selecting encryption and redundancy schemes per object?

Reply:
I think it makes sense to have an API for that when directly working with storj libraries. And that the `client` config (or maybe `bucket`) will define the defaults for it.